app/src/main/java/com/github/se/project/ui/components/LessonLocationDisplay.kt

```diff
@@ -23,33 +49,130 @@ fun LessonLocationDisplay(
+  LocationPermissionHandler { location ->
+    userLocation = location
+    isLocationChecked = true
+
+    // Update the camera position based on user location availability
+    cameraPositionState.position =
+        if (location != null) {
+          getCameraPositionForBothLocations(
+              userLocation = location, lessonLocation = lessonLocation)
+        } else {
+          CameraPosition.fromLatLngZoom(lessonLocation, 10f)
         }
   }
+
+  if (isLocationChecked) {
+    Column(modifier = modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
+      Text(text = "Lesson Location", style = MaterialTheme.typography.titleMedium)
+
+      Card(
+          modifier = Modifier.fillMaxWidth(),
+          elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)) {
+            Box(modifier = Modifier.fillMaxWidth().height(200.dp)) {
+              GoogleMap(
+                  modifier = Modifier.fillMaxSize(),
+                  cameraPositionState = cameraPositionState,
+                  properties = MapProperties(isMyLocationEnabled = (userLocation != null)),
+                  uiSettings =
+                      MapUiSettings(
+                          zoomControlsEnabled = false,
+                          scrollGesturesEnabled = false,
+                          zoomGesturesEnabled = false,
+                          tiltGesturesEnabled = false,
+                          rotationGesturesEnabled = false)) {
+                    // Marker for the lesson location
+                    Marker(state = MarkerState(position = lessonLocation), title = lessonTitle)
+                  }
+            }
+          }
+    }
+  }
+}
+
+/**
+ * Utility function to calculate the bounds to show both the user's location and lesson location,
+ * and update the camera position to fit them.
+ */
+fun getCameraPositionForBothLocations(
+    userLocation: LatLng?,
+    lessonLocation: LatLng,
+): CameraPosition {
+  // If we don't have the user's location, just focus on the lesson location
+  if (userLocation == null) {
+    return CameraPosition.fromLatLngZoom(lessonLocation, 10f)
+  }
+
+  // Otherwise, calculate the bounds to fit both locations
+  val latitudes = listOf(userLocation.latitude, lessonLocation.latitude)
+  val longitudes = listOf(userLocation.longitude, lessonLocation.longitude)
+
+  // Get the bounds to show both locations
+  val latMin = latitudes.minOrNull() ?: userLocation.latitude
+  val latMax = latitudes.maxOrNull() ?: userLocation.latitude
+  val lonMin = longitudes.minOrNull() ?: userLocation.longitude
+  val lonMax = longitudes.maxOrNull() ?: userLocation.longitude
+
+  // Create the bounds
+  val bounds = LatLngBounds(LatLng(latMin, lonMin), LatLng(latMax, lonMax))
+
+  // Calculate the zoom level based on the distance between the two locations
+  val distance = calculateDistance(userLocation, lessonLocation)
+  val zoomLevel = adjustZoomBasedOnDistance(distance)
```

Comment:
(This is a continuity of the comment in the adjustZoom function)
Here we could then check if the returned zoom is 0 and in this case replace the return value of the function by the default location (the one used when user doesn't allow to share his position)